jbmc/src/java_bytecode/java_bytecode_parser.cpp

```diff
@@ -1896,33 +1926,45 @@ java_bytecode_parsert::parse_method_handle(const method_handle_infot &entry)
   const name_and_type_infot &name_and_type =
     ref_entry.get_name_and_type(pool_entry_lambda);
 
-  std::string class_name = class_entry.get_name(pool_entry_lambda);
-  // replace '.' for '$' (inner classes)
-  std::replace(class_name.begin(), class_name.end(), '.', '$');
-  // replace '/' for '.' (package)
-  std::replace(class_name.begin(), class_name.end(), '/', '.');
-  const std::string method_ref =
-    class_name + "." + name_and_type.get_name(pool_entry_lambda) + ':' +
-    name_and_type.get_descriptor(pool_entry_lambda);
+  // The following lambda kinds specified in the JVMS (see for example
+  // https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-5.html#jvms-5.4.3.5
+  // but which aren't actually created by javac. Java has no syntax for a field-
+  // refernce like this, and even writing a stereotypical lambda like
+  // Producer<FieldType> = instance -> instance.field does not generate this
+  // kind of handle, but rather a synthetic method implementing the getfield
+  // operation.
+  // We don't implement a handle kind that can't be produced yet, but should
+  // they ever turn up this is where to fix them.
 
-  lambda_method_handlet lambda_method_handle;
-
-  if(has_prefix(name_and_type.get_name(pool_entry_lambda), "lambda$"))
+  if(
+    entry.get_handle_kind() ==
+      method_handle_infot::method_handle_kindt::REF_getField ||
+    entry.get_handle_kind() ==
+      method_handle_infot::method_handle_kindt::REF_putField ||
+    entry.get_handle_kind() ==
+      method_handle_infot::method_handle_kindt::REF_getStatic ||
+    entry.get_handle_kind() ==
+      method_handle_infot::method_handle_kindt::REF_putStatic)
   {
-    // names seem to be lambda$POSTFIX$NUM
-    // where POSTFIX is FUN for a function name in which the lambda is define
-    //                "static" when it is a static member of the class
-    //                "new" when it is a class variable, instantiated in <init>
-    lambda_method_handle.lambda_method_name =
-      name_and_type.get_name(pool_entry_lambda);
-    lambda_method_handle.lambda_method_ref = method_ref;
-    lambda_method_handle.handle_type =
-      method_handle_typet::LAMBDA_METHOD_HANDLE;
-
-    return lambda_method_handle;
```

Comment:
:bulb: Given that this could happen - it might make sense to output a warning, or even an invariant